lib/adapters/tool/testplane/index.ts

```diff
@@ -0,0 +1,146 @@
+import _ from 'lodash';
+import Testplane, {type Config, type TestCollection} from 'testplane';
+import type {CommanderStatic} from '@gemini-testing/commander';
+
+import {BaseToolAdapter, type BaseToolAdapterOptions, type UpdateReferenceOpts} from '../base';
+import {parseConfig} from '../../../config';
+import {HtmlReporter} from '../../../plugin-api';
+import {ApiFacade} from '../../../gui/api/facade';
+import {createTestRunner} from './runner';
+import {EventSource} from '../../../gui/event-source';
+import {GuiReportBuilder} from '../../../report-builder/gui';
+import {handleTestResults} from './test-results-handler';
+import {ToolName} from '../../../constants';
+
+import type {TestSpec, CustomGuiActionPayload} from '../types';
+import type {ReporterConfig, CustomGuiItem} from '../../../types';
+
+type TestplaneEnhanced = Testplane & { gui: ApiFacade, htmlReporter: HtmlReporter };
+
+type ReplModeOption = {
+    enabled: boolean;
+    beforeTest: boolean;
+    onFail: boolean;
+}
+
+export class TestplaneToolAdapter extends BaseToolAdapter {
+    private _tool: TestplaneEnhanced;
+    private _reporterConfig: ReporterConfig;
+    private _htmlReporter: HtmlReporter;
+
+    constructor(opts: BaseToolAdapterOptions) {
+        super(opts);
+
+        if ('tool' in opts) {
+            this._tool = opts.tool as TestplaneEnhanced;
+            this._reporterConfig = opts.reporterConfig;
+        } else {
+            // in order to not use static report with gui simultaneously
+            process.env['html_reporter_enabled'] = false.toString();
```

Comment:
I need to disable `html-reporter` plugin when it is called from `html-reporter` binary, because in this case I will have gui report and static report in the same time.